clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -302,11 +302,6 @@ private void throwIfOffsetOutOfRange() throws OffsetOutOfRangeException {
             throwIfOffsetOutOfRange();
 
             for (PartitionRecords<K, V> part : this.records) {
-                if (!subscriptions.isFetchable(part.partition)) {
```

Comment:
I think we still need this check. There may be a rebalance between the time a fetch response is handled and the time it is returned to the user.